internal/cache/metadata/stat_cache.go

```diff
@@ -51,19 +51,27 @@ type StatCache interface {
 	LookUp(name string, now time.Time) (hit bool, o *gcs.Object)
 }
 
-// Create a new stat cache that holds the given number of entries, which must
-// be positive.
-func NewStatCache(capacity int) (sc StatCache) {
-	sizeOfEntry := entry{}.Size()
-	sc = &statCache{
-		c: lru.NewCache(uint64(capacity) * sizeOfEntry),
+// Create a new bucket-view to the passed shared-cache object.
+func NewStatCacheBucketView(_sharedCache *lru.Cache, _bucketName string) StatCache {
```

Comment:
I couldn't reuse the names sharedCachee etc without bloating up more codelines, but also wanted to keep names similar to make it easier to read on the caller side. I haven't seen any convention in golang for such scenarios, so I took the convention from cpp. Anyway, I have now shortened the name instead of prefixing with underscore.